R/rlistings.R

```diff
@@ -435,3 +435,51 @@ add_listing_col <- function(df,
   df <- add_listing_dispcol(df, name)
   df
 }
+
+#' Split Listing by Values of a Variable
+#'
+#' @description `r lifecycle::badge("experimental")`
+#'
+#' Split is performed based on unique values of the given parameter present in the listing.
+#' Each listing can only be split by variable once. If this function is applied prior to
+#' pagination, parameter values will be separated by page.
+#'
+#' @param lsting listing_df. The listing to split.
+#' @param var character. Name of the variable to split on.
+#' @param page_prefix character. Prefix to be appended with the split value (`var` level),
+#'   at the end of the subtitles, corresponding to each resulting list element (listing).
+#'
+#' @return A list of `lsting_df` objects each corresponding to a unique value of `var`.
+#'
+#' @note This function should only be used after the complete listing has been created. The
+#'   listing cannot be modified further after applying this function.
+#'
+#' @examples
+#' dat <- ex_adae[1:20, ]
+#'
+#' lsting <- as_listing(
+#'     dat,
+#'     key_cols = c("USUBJID", "AGE"),
+#'     disp_cols = "SEX",
+#'     main_title = "title",
+#'     main_footer = "footer"
+#'   ) %>%
+#'   add_listing_col("BMRKR1", format = "xx.x") %>%
+#'   split_listing_by_var("SEX")
+#'
+#' lsting
+#'
+#' @export
+split_listing_by_var <- function(lsting, var, page_prefix = var) {
```

Comment:
as a function name would it be better like `paginate_by_var`? Otherwise the splitting could imply more layout-related features

Reply:
That may imply to people that it paginates to files - it really only splits the listing into a list according to the variable, which doesn't necessitate pagination, but I don't have a strong opinion either way :)

Reply:
maybe `set_page_by_var`?